Import all the important library needed

In [15]:
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np
import seaborn as sns
import random
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import StandardScaler
from sklearn.linear_model import LinearRegression
from sklearn.metrics import mean_squared_error
from sklearn.metrics import r2_score
from sklearn.metrics import mean_absolute_error


import the dataset and display the first 5 records of the daataset

In [2]:
df = pd.read_csv('yacht_hydro.csv')
df.head()

LC     PC   L/D  B/Dr   L/B     Fr    Rr
0 -2.3  0.568  4.78  3.99  3.17  0.125  0.11
1 -2.3  0.568  4.78  3.99  3.17  0.150  0.27
2 -2.3  0.568  4.78  3.99  3.17  0.175  0.47
3 -2.3  0.568  4.78  3.99  3.17  0.200  0.78
4 -2.3  0.568  4.78  3.99  3.17  0.225  1.18

As we can see that following attribute information <br>

1.Longitudinal position of the center of buoyancy [LC], adimensional. <br>
2.Prismatic coefficient [PC], adimensional.<br>
3.Length-displacement ratio [L/D], adimensional.<br>
4.Beam-draught ratio [B/Dr], adimensional.<br>
5.Length-beam ratio [L/B], adimensional.<br>
6.Froude number [Fr], adimensional.<br>

The measured variable is the residuary resistance per unit weight of displacement:<br>

1.Residuary resistance per unit weight of displacement [Rr], adimensional.<br>

Prediction of residuary resistance of sailing yachts at the initial design stage is of a great value for evaluating the ship’s performance and for estimating the required propulsive power.<br>

So Here our intention is to predict Rr From given data.


In [3]:
df.shape

(308, 7)

The Delft data set comprises 308 full-scale experiments, which were performed at the Delft Ship Hydromechanics Laboratory for that purpose.

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 308 entries, 0 to 307
Data columns (total 7 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   LC      308 non-null    float64
 1   PC      308 non-null    float64
 2   L/D     308 non-null    float64
 3   B/Dr    308 non-null    float64
 4   L/B     308 non-null    float64
 5   Fr      308 non-null    float64
 6   Rr      308 non-null    float64
dtypes: float64(7)
memory usage: 17.0 KB


In [5]:
df.isnull().sum()

LC      0
PC      0
L/D     0
B/Dr    0
L/B     0
Fr      0
Rr      0
dtype: int64

As we can observe following dataset does not contains any missing values.

# Target Setting

In [6]:
target = ['Rr']
dataY = df[target]
dataX = df.drop(target, axis=1)

In [8]:
df_columns = list(dataX.columns)
print(df_columns)

['LC', 'PC', 'L/D', 'B/Dr', 'L/B', 'Fr']


Now, <br>
dataX - contains all the columns except the target column <br>
dataY - contains the target column

In [11]:
dataX = np.array(dataX)
dataY = np.array(dataY)

In [22]:
lengthOfX = len(dataX)
print(lengthOfX)

M = list(range(m))
random.seed(2021)
random.shuffle(M)

308


In [23]:
trainX=dataX[M[0:(m//4)*3]]
trainY=dataY[M[0:(m//4)*3]]
testX=dataX[M[(m//4)*3:]]
testY=dataY[M[(m//4)*3:]]

This code splits the input data and corresponding labels into training and testing sets using a random shuffle. The division is such that 75% of the data is used for training and 25% for testing.  <br>

So, after this code is executed, the trainX and trainY arrays will contain 75% of the shuffled input data and labels, respectively, while the testX and testY arrays will contain the remaining 25%. These arrays can be used to train and evaluate a machine learning model


In [24]:
print(len(trainX))
print(len(testX))

231
77


In [25]:
train_df = pd.DataFrame(trainX)
test_df = pd.DataFrame(testX)

In [26]:
train_df.columns = df_columns
test_df.columns = df_columns

In [31]:
def create_numeric_feature(input_df):
    use_columns = df_columns
    return input_df[use_columns].copy()

So, the create_numeric_feature function creates a new DataFrame containing only a subset of columns specified in df_columns, which are assumed to be numeric features. This function can be used as a preprocessing step for machine learning models that require a subset of the columns in the input data to be numeric features.